internal/testsacc/alb_virtual_service_resource_test.go

```diff
@@ -0,0 +1,224 @@
+								enabled = true
+								pool_id = cloudavenue_alb_pool.example.id
+								virtual_ip = "192.168.10.10"
+								certificate_id = "urn:vcloud:certificateLibraryItem:f9caac3a-2555-477e-ae58-0740687d4daf"
+								service_type = "HTTPS"
+								service_ports = [
+									{
+										port_start = 443
+										port_type = "TCP_PROXY"
+										port_ssl = "true"
+									},
+									{
+										port_start = 8443
+									}
+								]
+							}`),
+						Checks: []resource.TestCheckFunc{
+							resource.TestCheckResourceAttr(resourceName, "name", testsacc.GetValueFromTemplate(resourceName, "name")),
+							resource.TestCheckResourceAttr(resourceName, "description", testsacc.GetValueFromTemplate(resourceName, "description")),
+							resource.TestCheckResourceAttrSet(resourceName, "edge_gateway_id"),
+							resource.TestCheckResourceAttr(resourceName, "enabled", "true"),
+							resource.TestCheckResourceAttr(resourceName, "certificate_id", "urn:vcloud:certificateLibraryItem:f9caac3a-2555-477e-ae58-0740687d4daf"),
+							resource.TestCheckResourceAttr(resourceName, "virtual_ip", "192.168.10.10"),
+							resource.TestCheckResourceAttr(resourceName, "service_type", "HTTPS"),
+							resource.TestCheckResourceAttr(resourceName, "service_ports.0.port_start", "443"),
+							resource.TestCheckResourceAttr(resourceName, "service_ports.0.port_type", "TCP_PROXY"),
+							resource.TestCheckResourceAttr(resourceName, "service_ports.0.port_ssl", "true"),
+							resource.TestCheckResourceAttr(resourceName, "service_ports.1.port_start", "8443"),
+							resource.TestCheckResourceAttr(resourceName, "service_ports.1.port_ssl", "false"),
+						},
+					},
+				},
+				// ! Imports testing
+				Imports: []testsacc.TFImport{
+					{
+						ImportStateIDBuilder: []string{"edge_gateway_id", "name"},
+						ImportState:          true,
+						ImportStateVerify:    true,
+					},
+					{
+						ImportStateIDBuilder: []string{"edge_gateway_name", "name"},
+						ImportState:          true,
+						ImportStateVerify:    true,
+					},
+				},
+				// ! Destroy testing
+				Destroy: true,
+			}
+		},
+		"example_with_public_ip": func(_ context.Context, resourceName string) testsacc.Test {
+			return testsacc.Test{
+				CommonChecks: []resource.TestCheckFunc{
+					resource.TestCheckResourceAttrSet(resourceName, "id"),
+					resource.TestCheckResourceAttrSet(resourceName, "name"),
+				},
+				// ! Create testing
+				// * Create a new Virtual Service with a public IP and service_type: L4 (TCP)
+				Create: testsacc.TFConfig{
+					TFConfig: testsacc.GenerateFromTemplate(resourceName, `
+						resource "cloudavenue_alb_virtual_service" "example_with_public_ip" {
+							name = {{ generate . "name" }}
+							description = {{ generate . "description" }}
+							edge_gateway_name = data.cloudavenue_edgegateway.example_with_id.name
+							enabled = true
+							pool_id = cloudavenue_alb_pool.example.id
+							virtual_ip = "62.161.20.174"
```

Comment:
Add comment to indicate that this is temporary and must be replaced by the resource or datasource once it is created